pkg/doc/verifiable/credential.go

```diff
@@ -220,19 +268,37 @@ func (ja JWTAlgorithm) Jose() jose.SignatureAlgorithm {
 
 // rawCredential is a basic verifiable credential
 type rawCredential struct {
-	Context        []string          `json:"@context,omitempty"`
+	Context        []interface{}     `json:"@context,omitempty"`
 	ID             string            `json:"id,omitempty"`
-	Type           []string          `json:"type,omitempty"`
+	Type           interface{}       `json:"type,omitempty"`
 	Subject        Subject           `json:"credentialSubject,omitempty"`
 	Issued         *time.Time        `json:"issuanceDate,omitempty"`
 	Expired        *time.Time        `json:"expirationDate,omitempty"`
 	Proof          *Proof            `json:"proof,omitempty"`
 	Status         *CredentialStatus `json:"credentialStatus,omitempty"`
 	Issuer         interface{}       `json:"issuer,omitempty"`
-	Schema         *CredentialSchema `json:"credentialSchema,omitempty"`
+	Schema         interface{}       `json:"credentialSchema,omitempty"`
+	Evidence       *Evidence         `json:"evidence,omitempty"`
+	TermsOfUse     []TermsOfUse      `json:"termsOfUse,omitempty"`
 	RefreshService *RefreshService   `json:"refreshService,omitempty"`
 }
 
+type typeSingle struct {
+	Type string `json:"type,omitempty"`
+}
+
+type typeMultiple struct {
+	Types []string `json:"type,omitempty"`
+}
+
+type credentialSchemaSingle struct {
```

Comment:
Could you explain a bit more why you needed an extra `credentialSchemaSingle` type?

Reply:
You are right.